hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -170,11 +170,18 @@ public Response getKeysForContainer(
       for (ContainerKeyPrefix containerKeyPrefix : containerKeyPrefixMap
           .keySet()) {
 
-        // Directly calling get() on the Key table instead of iterating since
-        // only full keys are supported now. When we change to using a prefix
-        // of the key, this needs to change to prefix seek.
-        OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable(getBucketLayout())
+        // Directly calling getSkipCache() on the Key/FileTable table
+        // instead of iterating since only full keys are supported now. We will
+        // try to get the OmKeyInfo object by searching the KEY_TABLE table with
+        // the key prefix. If it's not found, we will then search the FILE_TABLE
+        OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable(BucketLayout.LEGACY)
```

Comment:
Just pointing it out here that directory keys won't be considered in the case of FSO however I guess it should be okay as the endpoint name is "getKeysForContainers".  The change looks good otherwise.

Reply:
Thank you for bringing this to my attention. I understand that directory keys won't be considered in the case of FSO, but as you rightly pointed out, since the endpoint name is "getKeysForContainers" and we are concerned only with the keys that carry data size, I believe these change is still relevant and helpful for debugging purposes.